C/tasks/check_updates/check_updates.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Fledge Check Updates
+ *
+ * Copyright (c) 2024 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Devki Nandan Ghildiyal
+ */
+
+#include <check_updates.h>
+#include <logger.h>
+
+#include <cstdlib>
+#include <thread>
+#include <csignal>
+#include <fstream>
+#include <errno.h>
+
+using namespace std;
+
+volatile std::sig_atomic_t signalReceived = 0;
+std::string packageListFile = "/tmp/fledge.upgrade.list";
+
+static void signalHandler(int signal)
+{
+	signalReceived = signal;
+}
+
+
+/**
+ * Constructor for CheckUpdates
+ */
+CheckUpdates::CheckUpdates(int argc, char** argv) : FledgeProcess(argc, argv)
+{
+	std::string paramName;
+	paramName = getName();
+	m_logger = new Logger(paramName);
+	m_logger->info("CheckUpdates starting - parameters name :%s:", paramName.c_str() );
+	m_mgtClient = this->getManagementClient();
+
+}
+
+/**
+ * Destructor for CheckUpdates
+ */
+CheckUpdates::~CheckUpdates()
+{
+}
+
+/**
+ * CheckUpdates run method, called by the base class to start the process and do the actual work.
+ */
+void CheckUpdates::run()
+{
+	// We handle these signals, add more if needed
+	std::signal(SIGINT,  signalHandler);
+	std::signal(SIGSTOP, signalHandler);
+	std::signal(SIGTERM, signalHandler);
+
+
+	if (!m_dryRun)
+	{
+		raiseAlerts();
+	}
+	processEnd();
+}
+
+/**
+ * Execute the raiseAlerts, create an alert for all the packages for which update is available
+ */
+void CheckUpdates::raiseAlerts()
+{
+	m_logger->info("raiseAlerts running");
+	try
+	{
+		for (auto key: getUpgradablePackageList())
+		{
+			std::string message = "A newer version of " + key + " is available for upgrade";
+			std::string urgency = "normal";
+			if (!m_mgtClient->raiseAlert(key,message,urgency))
+			{
+				m_logger->error("Failed to raise an alert for key=%s,message=%s,urgency=%s", key.c_str(), message.c_str(), urgency.c_str());
+			}
+		}
+		std::string command = "sudo rm -f " + packageListFile;
```

Comment:
This is very dangerous. I'm not happy about having to allow FogLAMP to run rm -rf as root. Can we please look at using open as I feel this will probably cause all sorts of issues when end users do security scans on FogLAMP.

Reply:
@MarkRiddoch : Since file is being created at /tmp/ and overwitten everytime. Can we leave file on as it is or use `sudo unlink <fileName>` command instead ?

Reply:
Fixed